### PR TITLE
feat(wip) : teams and ticket type in settings

### PR DESCRIPTION
**Pull request by esafwan**
- Added Teams (agent group) and ticket type in settings sidebar. 
- Can add, delete and update ticket types. 
- In teams, we  can add new teams and tag multiple users against it. 
- These users will get notified if the ticket status go overdue (being implemented).

### CODE REVIEW

File: desk/src/components/desk/global/AddNewTeamDialog.vue [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+							type="text"
+							placeholder="Product Expert"
+							v-model="teamName"
+						/>
+					</div>
+					<div>
+						<label class="block text-sm leading-4 text-gray-700"
+							>Users</label
+						>
+						<VueMultiselect
```
Comment (kamaljohnson) [inline desk/src/components/desk/global/AddNewTeamDialog.vue]: @fadilsid if you want you could take reference from the `Add Agent Dialog` which has a similar input

image

The component is currently hardcoded for adding new agents only, but you could take this as a reference

**Comment by kamaljohnson** [comment]
the settings sidebar keeps highlighting `Agents` setting if `Teams` or `Ticket Type` is clicked.
https://user-images.githubusercontent.com/22856401/210192255-6b1d1f7e-a295-4882-b3ba-87a5b29b2695.mov

**Comment by fadilsid** [comment]
> the settings sidebar keeps highlighting `Agents` setting if `Teams` or `Ticket Type` is clicked.
> Screen.Recording.2023-01-02.at.8.27.37.AM.mov

fixed in [38aaf4f](https://github.com/frappe/desk/pull/888/commits/38aaf4f533e546749bc20bce3de93a21eccb95d7)


https://user-images.githubusercontent.com/73826691/210218696-c4869edf-f33c-4a9a-ad35-a5524883b860.mp4

**Comment by kamaljohnson** [comment]
image

@fadilsid, please add use teamDoc?.users instead of teamDoc.users
this is showing error since teamDoc will only be set to doc after fetching the resources

**Comment by kamaljohnson** [comment]
@fadilsid please a validation check before adding the email str to the list

https://user-images.githubusercontent.com/22856401/210308739-1dc35d8f-e373-48a6-b143-ea7fb922b3ad.mov

**Comment by kamaljohnson** [comment]
@fadilsid, please add use ticketTypeDoc?.priority instead of ticketTypeDoc.priority, reason same as described above

image

**Comment by kamaljohnson** [comment]
@fadilsid also there are formatting issues too. fix them too

**Comment by fadilsid** [comment]
> @fadilsid please a validation check before adding the email str to the list
> Screen.Recording.2023-01-03.at.11.53.41.AM.mov

fixed email validation in [a6763f7](https://github.com/frappe/desk/pull/888/commits/a6763f71358370e2ad6f957d4e8f531d64dfebd6)

https://user-images.githubusercontent.com/73826691/210320208-d7d7e003-f06f-4825-a5c8-5baf600f8bbb.mp4

**Pull request merged**